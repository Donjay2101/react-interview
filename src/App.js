import React from "react";
//import "./App.css";

import RouteComponent from "./components/route/route";
import HeaderComponent from "./components/header/header";

import Aux from "./components/hoc/Auxiliary";

function App() {
  return(
    <Aux>
        <div className="container">
        <HeaderComponent />
        <RouteComponent />
        </div>
    </Aux>
  ) 
  
  
}

export default App;
