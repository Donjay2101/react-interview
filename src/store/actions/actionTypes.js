export const GET_ORDERINFORMATION = "GET_ORDERiNFORMATION";
export const UPDATE_DELIVERYTYPE = "UPDATE_DELIVERYTYPE";
export const ADD_PIZZA = "ADD_PIZZA";
export const REMOVE_PIZZA = "REMOVE_PIZZA";
export const UPDATE_PIZZA = "UPDATE_PIZZA";
export const PROCESS_ORDER = "PROCESS_ORDER";


