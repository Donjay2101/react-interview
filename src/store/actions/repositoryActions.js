import * as actionTypes from "./actionTypes";
// import axios from "../../axios/axios";

//order information to get from state

/*---------------------------------
  Get Information from store
---------------------------------*/
const getOrderInformationSuccess = () => {
  return {
    type: actionTypes.GET_ORDERINFORMATION   
  };
};

export const getOrderInformation = () => {
  return dispatch => {    
    dispatch(getOrderInformationSuccess());
  }
};

/*---------------------------------
  Get Information from store
---------------------------------*/


/*---------------------------------
  update  Information in store
---------------------------------*/

//-----update Delivery Type
const updateDeliveryTypeSuccess = response => {
  return {
    type: actionTypes.UPDATE_DELIVERYTYPE,
    data: response
  };
};
export const updateDeliveryType=(obj)=>{
  return dispatch => {
    dispatch(updateDeliveryTypeSuccess(obj));
    }

}
//-----update Delivery Type
const addPizzaSuccess = response => {
  return {
    type: actionTypes.ADD_PIZZA,
    data: response
  };
};
export const addPizza=(obj)=>{
  return dispatch => {
    dispatch(addPizzaSuccess(obj));
    }
}

const deletePizzaSuccess = response => {
  return {
    type: actionTypes.REMOVE_PIZZA,
    data: response
  };
};
export const deletePizza=(obj)=>{
  return dispatch => {
    dispatch(deletePizzaSuccess(obj));
    }

}

const updatePizzaSuccess = response => {
  return {
    type: actionTypes.UPDATE_PIZZA,
    data: response
  };
};


export const updatePizza=(obj)=>{
  return dispatch => {
    dispatch(updatePizzaSuccess(obj));
    }
}



const processOrderSuccess = () => {
  return {
    type: actionTypes.PROCESS_ORDER
  };
};

export const processOrder=()=>{
  return dispatch => {
    dispatch(processOrderSuccess());
    }
}








//code to use axios to get data from API.
/*
  axios
      .delete(url)
      .then(response => {
        dispatch(deleteDataSuccess(response.data));
      })
      .catch(error => {
        //TODO: handle the error when implemented
      });
*/