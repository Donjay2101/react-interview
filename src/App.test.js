import React from 'react';
import { Provider } from 'react-redux';
import thunk from "redux-thunk";
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import  * as actions  from "./store/actions/repositoryActions";
 
import App from './App';
 
const mockStore = configureStore([thunk]);
 
describe('My Connected React-Redux Component', () => {
  let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
      deliveryType:'',
      pizzas:[]
    });
 
    store.dispatch = jest.fn();
 
    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
 
  it('should dispatch an action selection of delivery type option', () => {    
    renderer.act(() => {
      var c = component.root.findAllByProps({name:'deliveryType'});
      c[0].props.onChange({target:{value:"delivery"}});
    });
    
    expect(store.dispatch).toHaveBeenCalledTimes(1);  
  });
});