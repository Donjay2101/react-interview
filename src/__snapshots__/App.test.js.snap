// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron"
  >
    <h2>
      Rock on Pizza Delivery
    </h2>
    <p>
      Welcome Rock On Pizza delivery.
    </p>
  </div>
  <div
    className="row deliveryType"
  >
    <div
      className="col-md-6 col-sm-1"
    >
      <label
        className="card"
      >
        <input
          checked={false}
          className="radio"
          name="deliveryType"
          onChange={[Function]}
          readOnly={true}
          type="radio"
          value="delivery"
        />
        <span>
          Delivery
        </span>
      </label>
    </div>
    <div
      className="col-md-6 col-sm-1"
    >
      <label
        className="card"
      >
        <input
          checked={false}
          className="radio"
          name="deliveryType"
          onChange={[Function]}
          readOnly={true}
          type="radio"
          value="pickup"
        />
        <span>
          Pick Up
        </span>
      </label>
    </div>
  </div>
</div>
`;
